### ISSUE TITLE
DNS: Can not create rrset ressources with long names

### ISSUE THREAD

**Issue by Slm0n87** [stackitcloud/terraform-provider-stackit#344]
Currently DNS rrset ressources with long names can not be created by terraform:

```
terraform apply
╷
│ Error: Invalid Attribute Value Length
│ 
│   with stackit_dns_record_set.somethigveryveryveryveryveryverylong-221-some-domain.test,
│   on stackit-cloud.tf line 1127, in resource "stackit_dns_record_set" "somethigveryveryveryveryveryverylong-221-some-domain.test":
│ 1127:   name       = "somethigveryveryveryveryveryverylong-221.scanner.some-domain.test."
│ 
│ Attribute name string length must be at most 63, got: 66`
```

**Comment by DiogoFerrao** [comment]
Thank you for your report @Slm0n87!
I will investigate on this issue and provide an update soon.

**Comment by DiogoFerrao** [comment]
I investigated on this issue and it seems to be a problem with the API. Even though the current [API docs](https://docs.api.eu01.stackit.cloud/documentation/dns/version/v1#tag/record-set/operation/post-rrset) claim that `name` can have up to 253 characters, this is not true. When I tried calling the API directly using Postman, it provides an error, which is the error we are beautifying in the provider (name can't have more than 63 characters).
![image](https://github.com/stackitcloud/terraform-provider-stackit/assets/33052643/0aefb392-1440-4a69-9182-41fd14387780)

**Comment by Slm0n87** [comment]
Thanks @DiogoFerrao to pick this issue up quick.  
I think the details of my explanation were not detailed enough.  
As the description in the API docs you have linked to mentiones, the rrset name has to match rfc1035 Section 2.3.4, so:
```
[2.3.4] Size limits

Various objects and parameters in the DNS have size limits.  They are
listed below.  Some could be easily changed, others are more
fundamental.

labels          63 octets or less

names           255 octets or less
...
```
So within your test to have to create a rrset name with max. 63 character labels (the part between the dots) in addition to the max. 253 character limit (we removed 2 chars for the rrset name, because of the zone name + trailing dot.

**Comment by DiogoFerrao** [comment]
This issue was addressed in Release [v0.16.0](https://github.com/stackitcloud/terraform-provider-stackit/releases/tag/v0.16.0)
